.webpack/webpack.prod.js

```diff
@@ -1,22 +1,96 @@
 /* global __dirname module */
 
 /*
-This configuration should be used for production installs.
-It is the default webpack configuration.
+Production webpack configuration.
 */
-const path = require('path');
 const webpack = require('webpack');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
 const { merge } = require('webpack-merge');
 
-const common = require('./webpack.common');
-const projectRootDir = path.resolve(__dirname, '..');
+const TerserPlugin = require('terser-webpack-plugin');
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
+
+const base = require('./webpack.base');
 
-module.exports = merge(common, {
+module.exports = merge(base, {
+  devtool: 'source-map',
   mode: 'production',
+  module: {
+    rules: [
+      {
+        test: /\.(sc|sa|c)ss$/,
+        use: [
+          {
+            loader: MiniCssExtractPlugin.loader
+          },
+          {
+            loader: 'css-loader',
+            options: {
+              sourceMap: false
+            }
+          },
+          {
+            loader: 'resolve-url-loader'
+          },
+          {
+            loader: 'postcss-loader',
+            options: {
+              sourceMap: true
+            }
+          },
+          {
+            loader: 'sass-loader',
+            // resolve-url-loader needs all loaders below its decl to have sourcemaps enabled
+            options: { sourceMap: true }
+          }
+        ]
+      }
+    ]
+  },
   plugins: [
     new webpack.DefinePlugin({
       __OPENMCT_ROOT_RELATIVE__: '""'
+    }),
+    new MiniCssExtractPlugin({
+      filename: '[name].css',
+      chunkFilename: '[name].css'
     })
   ],
-  devtool: 'source-map'
+  // TODO (@evenstensberg): splitchunks when application is aligned towards it
+  optimization: {
+    runtimeChunk: false,
+    minimizer: [
+      new TerserPlugin({
+        terserOptions: {
+          output: {
+            comments: false
+          },
+          compress: {
+            passes: 3,
+            pure_getters: true,
+            unsafe: true
+          },
+          ecma: undefined,
+          warnings: false,
+          parse: {
+            html5_comments: false
+          },
+          mangle: true,
+          module: false,
+          toplevel: false,
+          nameCache: null,
+          ie8: false,
+          keep_classnames: false,
+          keep_fnames: false,
+          safari10: false
```

Comment:
We need to be very careful here as we see more e2e test flakes and odd behavior with some of these options.

What's the least aggressive we can be with minimization optimization while also shrinking the bundle size as much as possible? In my experiments I found we should be able to shrink `openmct.js` from ~35MB to ~5MB. If we can get close to that with mostly default settings, we should opt for that.

Reply:
I think that you'll need to manually test each of these settings, and enable/disable yourself to customize the fine-grained optimizations to your build.